File: ios/ReservationApp/ReservationApp/Presentation/Views/Admin/Menu/Components/AdminMealCardView.swift
//
//  AdminMealCardView.swift
//  ReservationApp
//
//  Created by Mehmet Ali Kısacık on 13.11.2025.
//

import SwiftUI

struct AdminMealCardView: View {
    let meal: Meal
    let onEdit: () -> Void
    let onDelete: () -> Void
    
    var body: some View {
        VStack(spacing: 0) {
            // Image Section
            ZStack(alignment: .topTrailing) {
                // Meal Image
                CachedAsyncImage(url: URL(string: meal.imageUrl ?? "")) { image in
                    image
                        .resizable()
                        .aspectRatio(contentMode: .fill)
                } placeholder: {
                    LinearGradient(
                        colors: [AppColors.primaryLight.opacity(0.3), AppColors.primaryMain.opacity(0.3)],
                        startPoint: .topLeading,
                        endPoint: .bottomTrailing
                    )
                }
                .frame(height: 200)
                .frame(maxWidth: .infinity)
                .clipped()
                
                // Category Badge
                CategoryBadge(name: meal.categoryName)
                    .padding(ThemeManager.shared.spacing.sm)
            }
            
            // Content Section
            VStack(alignment: .leading, spacing: ThemeManager.shared.spacing.sm) {
                // Name and Price Row
                HStack(alignment: .top) {
                    VStack(alignment: .leading, spacing: ThemeManager.shared.spacing.xs) {
                        Text(meal.name)
                            .font(AppTypography.h6())
                            .fontWeight(.semibold)
                            .foregroundColor(AppColors.textPrimary)
                            .lineLimit(2)
                        
                        Text(meal.restaurantName)
                            .font(AppTypography.body2())
                            .foregroundColor(AppColors.textSecondary)
                    }
                    
                    Spacer()
                    
                    if let price = meal.price {
                        Text(formatPrice(price))
                            .font(AppTypography.h6())
                            .fontWeight(.semibold)
                            .foregroundColor(AppColors.textPrimary)
                    }
                }
                
                // Action Buttons
                HStack(spacing: ThemeManager.shared.spacing.sm) {
                    Button(action: onEdit) {
                        HStack {
                            Image(systemName: "pencil")
                            Text("Düzenle")
                        }
                        .font(AppTypography.body2())
                        .foregroundColor(AppColors.textPrimary)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, ThemeManager.shared.spacing.sm)
                        .background(Color.clear)
                        .overlay(
                            RoundedRectangle(cornerRadius: ThemeManager.shared.borderRadius.button.value)
                                .stroke(AppColors.borderDefault, lineWidth: 1)
                        )
                        .cornerRadius(ThemeManager.shared.borderRadius.button.value)
                    }
                    
                    Button(action: onDelete) {
                        Image(systemName: "trash")
                            .foregroundColor(.white)
                            .frame(width: 48, height: 48)
                            .background(AppColors.errorMain)
                            .cornerRadius(ThemeManager.shared.borderRadius.button.value)
                    }
                }
            }
            .padding(ThemeManager.shared.spacing.md)
        }
        .frame(maxWidth: .infinity)
        .background(AppColors.backgroundPaper)
        .cornerRadius(ThemeManager.shared.borderRadius.card.value)
        .shadow(color: Color.black.opacity(0.05), radius: 4, x: 0, y: 2)
    }
    
    private func formatPrice(_ price: Double) -> String {
        let formatter = NumberFormatter()
        formatter.numberStyle = .decimal
        formatter.minimumFractionDigits = 2
        formatter.maximumFractionDigits = 2
        return "\(formatter.string(from: NSNumber(value: price)) ?? "\(price)")₺"
    }
}

File: android/app/src/main/java/com/reservationapp/ui/admin/menu/AdminMealCardView.kt
package com.reservationapp.ui.admin.menu

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material.icons.filled.Edit
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import coil.compose.AsyncImage
import com.reservationapp.core.ui.theme.*
import com.reservationapp.domain.model.Meal
import java.text.NumberFormat
import java.util.*

@Composable
fun AdminMealCardView(
    meal: Meal,
    onEdit: () -> Unit,
    onDelete: () -> Unit,
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = BackgroundPaper
        ),
        elevation = CardDefaults.cardElevation(defaultElevation = 2.dp)
    ) {
        Column(
            modifier = Modifier.fillMaxWidth()
        ) {
            // Image Section
            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(200.dp)
                    .clip(RoundedCornerShape(topStart = 12.dp, topEnd = 12.dp))
            ) {
                val imageUrl = meal.imageUrl
                if (!imageUrl.isNullOrEmpty()) {
                    AsyncImage(
                        model = imageUrl,
                        contentDescription = meal.name,
                        modifier = Modifier.fillMaxSize(),
                        contentScale = ContentScale.Crop
                    )
                } else {
                    Box(
                        modifier = Modifier
                            .fillMaxSize()
                            .background(
                                Brush.linearGradient(
                                    colors = listOf(
                                        PrimaryLight.copy(alpha = 0.3f),
                                        PrimaryMain.copy(alpha = 0.3f)
                                    )
                                )
                            )
                    )
                }

                // Category Badge
                meal.categoryName?.let { categoryName ->
                    Surface(
                        modifier = Modifier
                            .align(Alignment.TopEnd)
                            .padding(Spacing.sm),
                        shape = RoundedCornerShape(8.dp),
                        color = PrimaryLight
                    ) {
                        Text(
                            text = categoryName,
                            modifier = Modifier.padding(horizontal = Spacing.sm, vertical = Spacing.xs),
                            style = MaterialTheme.typography.labelSmall,
                            color = androidx.compose.ui.graphics.Color.White,
                            fontWeight = FontWeight.Medium
                        )
                    }
                }
            }

            // Content Section
            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .padding(Spacing.md),
                verticalArrangement = Arrangement.spacedBy(Spacing.sm)
            ) {
                // Name and Price Row
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.Top
                ) {
                    Column(
                        modifier = Modifier.weight(1f),
                        verticalArrangement = Arrangement.spacedBy(Spacing.xs)
                    ) {
                        Text(
                            text = meal.name,
                            style = MaterialTheme.typography.titleMedium.copy(
                                fontWeight = FontWeight.SemiBold
                            ),
                            color = TextPrimary,
                            maxLines = 2
                        )
                        Text(
                            text = meal.restaurantName ?: "",
                            style = MaterialTheme.typography.bodyMedium,
                            color = TextSecondary
                        )
                    }

                    meal.price?.let { price ->
                        Text(
                            text = formatPrice(price),
                            style = MaterialTheme.typography.titleMedium.copy(
                                fontWeight = FontWeight.SemiBold
                            ),
                            color = TextPrimary
                        )
                    }
                }

                // Action Buttons
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(Spacing.sm)
                ) {
                    OutlinedButton(
                        onClick = onEdit,
                        modifier = Modifier.weight(1f),
                        colors = ButtonDefaults.outlinedButtonColors(
                            contentColor = TextPrimary
                        ),
                        border = androidx.compose.foundation.BorderStroke(1.dp, BorderDefault)
                    ) {
                        Icon(
                            imageVector = Icons.Default.Edit,
                            contentDescription = null,
                            modifier = Modifier.size(18.dp)
                        )
                        Spacer(modifier = Modifier.width(Spacing.xs))
                        Text("Düzenle")
                    }

                    Button(
                        onClick = onDelete,
                        colors = ButtonDefaults.buttonColors(
                            containerColor = ErrorMain
                        ),
                        modifier = Modifier.size(48.dp),
                        contentPadding = PaddingValues(0.dp)
                    ) {
                        Icon(
                            imageVector = Icons.Default.Delete,
                            contentDescription = "Sil",
                            modifier = Modifier.size(24.dp)
                        )
                    }
                }
            }
        }
    }
}

private fun formatPrice(price: Double): String {
    val formatter = NumberFormat.getNumberInstance(Locale.getDefault())
    formatter.minimumFractionDigits = 2
    formatter.maximumFractionDigits = 2
    return "${formatter.format(price)}₺"
}

